update: server host shutdown and client disconnect (#1149)
Unity-Technologies/com.unity.multiplayer.docs: docs/components/networkmanager.md

## Disconnecting and Shutting Down

> ### Disconnecting

Reviewer: ```suggestion
### Disconnect from a network
```

> Disconnecting is rather simple, but you can't use/access any subsystems (that is, `NetworkSceneManager`) once the `NetworkManager` is stopped because they will no longer be available.  For client, host, or server modes, you only need to call the `NetworkManager.Shutdown` method as it will disconnect while shutting down.  

Reviewer: ```suggestion
When you disconnect from a network you can't use or access any subsystems, for example `NetworkSceneManager`, because they become unavailable when `NetworkManager` stops. For client, host, or server modes, call the `NetworkManager.Shutdown` method to disconnect and shut down at the same time.  
```

}
```

> ### Shutting Down

Reviewer: ```suggestion
### Shut down a network
```

Author: Shutting Down NetworkManager?

Shut down a network seems like you are shutting down an entire network.

Author: Possibly:
Shutting down a network session

> When you invoked `NetworkManager.Shutdown` any existing transport connections will be closed. The shutdown sequence for a client will complete by the next player loop update/frame. However, the shutdown sequence can take a bit longer for a server or host (server-host). In order to assure all clients are notified when a server-host is shutting down, upon invoking `NetworkManager.Shutdown` the server-host will first send out a disconnect notification to all currently connected clients and will populate the NetworkManager.Disconnect reason with one of the two following messages (depending upon whether it is a server or host instance):

Reviewer: ```suggestion
When you invoke `NetworkManager.Shutdown` it closes any existing transport connections. The shutdown sequence for a client finishes before the next player loop update or frame. However, the shutdown sequence can take a bit longer for a server or host (server-host). 
The server-host notifies all clients when it shuts down. To do this, when you invoke `NetworkManager.Shutdown` the server-host sends out a disconnect notification to all currently connected clients and populates the `NetworkManager.Disconnect` reason with one of the two following messages, depending upon whether it is a server or host instance:
```

- "Disconnected due to server shutting down."
- "Disconnected due to host shutting down."

> The server-host will attempt to wait for all client connections to close prior to finishing the shutdown sequence. This additional step on the server-host side can help to assure clients receive a disconnect notification and will help to prevent the client transport connection from timing out. Under the scenario where not all client connections are closed within 5 seconds after invoking `NetworkManager.Shutdown`, the server-host will automatically finish its shutdown sequence and any remaining connections will be closed.

Reviewer: ```suggestion
The server-host attempts to wait for all client connections to close before it finishes the shutdown sequence. This additional step on the server-host side helps to assure that clients receive a disconnect notification and prevents the client transport connection from timing out.  If some client connections take longer than five seconds to close after `NetworkManager.Shutdown` is invoked, the server-host automatically finishes the shutdown sequence and closes any remaining connections.
```

### Client Disconnection Notifications

Both the client and the server can subscribe to the `NetworkManager.OnClientDisconnectCallback` event to be notified when a client is disconnected.

**When disconnect notifications are triggered:**
- Clients are notified when they're disconnected by the server.

> - The server is notified when any client disconnects (whether the server disconnects the client or the client disconnects itself).

Reviewer: ```suggestion
- The server is notified when any client disconnects from the server, whether the server disconnects the client or the client disconnects itself.
```

- Both the server and clients are notified when their network connection is unexpectedly disconnected (network interruption)

**Client notification identifiers**

> - On the server-side, the client identifier parameter will always be the identifier of the client disconnected

Reviewer: ```suggestion
- On the server-side, the client identifier parameter is the identifier of the client that disconnects.
```

> - On the client-side, the client identifier parameter will always be the identifier assigned to the client

Reviewer: ```suggestion
- On the client-side, the client identifier parameter is the identifier assigned to the client.
```

>   - _The exception to this is when a client is disconnected before its connection is approved._

Reviewer: What happens in this exceptional case? Is there no identifier at all?
